src/go/pt-k8s-pxc-recovery/recover/recover.go

```diff
@@ -0,0 +1,326 @@
+package recover
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/percona/percona-toolkit/src/go/pt-k8s-pxc-recovery/kubectl"
+)
+
+type Cluster struct {
+	Name          string
+	Size          int
+	MostRecentPod string
+	Namespace     string
+}
+
+func (c *Cluster) SetClusterSize() error {
+	args := []string{
+		"get",
+		"pxc",
+		c.Name,
+		"-o",
+		"jsonpath='{.spec.pxc.size}'",
+	}
+	strSize, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return err
+	}
+	strSize = strings.Trim(strSize, "'")
+	c.Size, err = strconv.Atoi(strSize)
+	if err != nil {
+		return fmt.Errorf("error getting cluster size, %s", err)
+	}
+	return nil
+}
+
+func (c *Cluster) GetClusterImage() (string, error) {
+	args := []string{
+		"get",
+		"pod",
+		c.Name + "-pxc-0",
+		"-o",
+		"jsonpath='{.spec.containers[0].image}'",
+	}
+	clusterImage, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return "", fmt.Errorf("Error getting cluster image %s", err)
+	}
+	clusterImage = strings.Trim(clusterImage, "'")
+	return clusterImage, nil
+}
+
+func (c *Cluster) getPods() ([]string, error) {
+	args := []string{
+		"get",
+		"pods",
+		"--no-headers",
+		"-o",
+		"custom-columns=:metadata.name",
+	}
+	out, err := kubectl.RunCmd(c.Namespace, args...)
+	if err != nil {
+		return []string{}, err
+	}
+	formatedOutput := strings.Split(out, "\n")
+	podNames := []string{}
+	for _, podName := range formatedOutput {
+		if strings.Contains(podName, c.Name) && strings.Contains(podName, "pxc") {
+			podNames = append(podNames, podName)
+		}
+	}
+	return podNames, nil
+}
+
+func (c *Cluster) ConfirmCrashedStatus() error {
+	podNames, err := c.getPods()
+	if err != nil {
+		return fmt.Errorf("Error getting pods : %s", err)
+	}
+	for _, pod := range podNames {
+		logs, err := kubectl.RunCmd(c.Namespace, "logs", pod)
+		if err != nil {
+			return fmt.Errorf("error confirming crashed cluster status %s", err)
+		}
+		if !strings.Contains(logs, "grastate.dat") && !strings.Contains(logs, "safe_to_bootstrap") &&
+			!strings.Contains(logs, "It may not be safe to bootstrap the cluster from this node") {
+			return fmt.Errorf("found one or more pods in healthy state, can't use recovery tool, please restart failed pods manually")
```

Comment:
Please see point 4 in doc https://www.percona.com/doc/kubernetes-operator-for-pxc/recovery.html

We can not say the user "please do something manually" for the tool which should automatically  recover the cluster.
So, you need automatically kill the 'mysqld-ps' process if it is running.